### PR TITLE
[Merged by Bors] - feat(RingTheory): `Valuation.Integers.not_denselyOrdered_of_isPrincipalIdealRing`

### PR DESCRIPTION
**Pull request by pechersky**
On the way to the iff when the range of valuation on nonzeros is `MulArchimedean`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Valuation/Integers.lean [hunk 1]
```diff
@@ -144,6 +144,68 @@ theorem eq_algebraMap_or_inv_eq_algebraMap (hv : Integers v O) (x : F) :
...
+  simp only [Set.mem_inter_iff, Set.mem_Iio, Set.mem_range, Function.comp_apply] at this
+  obtain ⟨hx', y, rfl⟩ := this
+  rw [← v.map_one] at hx'
+  obtain ⟨⟨z, hzv⟩, hz, hz'⟩ := H.dense ⟨v (algebraMap O F y), Set.mem_range_self _⟩
+    ⟨v 1, Set.mem_range_self _⟩ hx'
+  simp only [Set.mem_range] at hzv
+  obtain ⟨z, rfl⟩ := hzv
+  have := hx.right
+  rw [mem_upperBounds] at this
+  refine (this _ ?_).not_lt hz
```
Comment (YaelDillies) [inline Mathlib/RingTheory/Valuation/Integers.lean:198]:
```suggestion
  refine (hx.right ?_).not_lt hz
```

**Review by YaelDillies — COMMENTED** [review]
Looks good to me

File: Mathlib/RingTheory/Valuation/Integers.lean (hunk 1 above)
Comment (pechersky) [reply Mathlib/RingTheory/Valuation/Integers.lean:198]: reverted -- I use `this` again below.

**Comment by YaelDillies** [comment]
I pushed a golf. Am happy with the rest

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**